### PR TITLE
Improve command handling testing

### PR DESCRIPTION
**Pull request by mattclarke**
### Issue
N/A

### Description of work
Splits out some of the command handling into a separate class.
It's a step in the right direction, but there is still more to come.

### Nominate for Group Code Review

- [ ] Nominate for code review

### CODE REVIEW

File: src/CommandParser.h [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+  /// \brief Extract the command name from the command.
+  ///
+  /// Note: the command is converted to lower-case.
+  ///
+  /// \param JSONCommand The JSON Command.
+  /// \return The command name.
+  static std::string extractCommandName(const nlohmann::json &JSONCommand);
+
+  static std::string const StopCommand;
```
Comment (SkyToGround) [inline src/CommandParser.h]: Minor thing: Could these be private?

File: src/CommandParser.h [hunk 2]
```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: BSD-2-Clause
```
Comment (SkyToGround) [inline src/CommandParser.h]: It seems to me that having `CommandParser` as a class is not really justified. The only shared state is the pointer to the logger. I think that the implementation would be slightly cleaner (and easier to test) by making `CommandParser` into a namespace. The pointer to the logger could still be retained as a shared state by removing it from the header file and replacing it with `static SharedLogger Logger = getLogger();` in the *.cpp-file.

Comment (mattclarke) [reply src/CommandParser.h]: Agreed, that was something I was thinking off doing then forgot!

Comment (mattclarke) [reply src/CommandParser.h]: I'll see how the next set of changes go and see if they fit with this.

**Pull request merged**